### PR TITLE
[FIX] core: flush env after pre upgrade scripts

### PR DESCRIPTION
**Pull request by Xavier-Do**
In some case, a pre migration script will call `xml_import.parse`
to force reload a no-update data in pre script. It is possible that
field will be marked as recompute in the process. The problem is
that the reference to the field added in `env.all.tocompute` won't
be the same as the reference of the same field after calling
setup_models, leading to an infinite loop while recomputing this field
since `fields.__get__` wont find self in `env.all.tocompute`.

This problem was discorvered trying to migrate a database with all modules
installed (no demo data) from 12.0 to 13.0 (for commit references:
http://runbot.odoo.com/runbot/build/1176576 with database comming from
http://runbot.odoo.com/runbot/build/1171714).

This commit add a check before executing `setup_models` in order to log if
some tocompute remains before breaking fields references
and add a flush after pre scipts to fix the current issue.

### CODE REVIEW

File: odoo/modules/registry.py [hunk 1]
```diff
@@ -231,6 +231,9 @@ def setup_models(self, cr):
         lazy_property.reset_all(self)
         env = odoo.api.Environment(cr, SUPERUSER_ID, {})
 
+        if env.all.tocompute:
```
Comment (Xavier-Do) [inline odoo/modules/registry.py]: I wonder if we shouldn't flush here as a one line fix instead

Comment (rco-odoo) [reply odoo/modules/registry.py]: You don't know whether the registry is in a correct state at this point.
Blindly flushing is unsafe and would simply hide a programming error.

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

**Comment by rco-odoo** [comment]
@robodoo r-

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**